dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/openaire/OpenAireExportUtil.java

```diff
@@ -1577,4 +1540,68 @@ private static boolean isValidEmailAddress(String email) {
         }
         return result;
     }
+
+    private static void writeRightsUriLicenseInfoAttribute(XMLStreamWriter xmlw, List<FileDTO> files, String language) throws XMLStreamException {
+        if(!Lists.isEmpty(files)) {
+            writeRightsHeader(xmlw, language);
+
+            if(areAllFilesRestricted(files)) {
+                xmlw.writeCharacters("Access to all files in the dataset is restricted.");
+            } else if(areAllFilesAllRightsReserved(files)) {
+                xmlw.writeCharacters("All rights reserved.");
+            } else if(areAllFilesUnderSameLincese(files)) {
+                xmlw.writeAttribute("rightsURI", files.get(0).getLicenseUrl());
+                xmlw.writeCharacters(files.get(0).getLicenseName());
+            } else if(hasRestrictedFile(files)) {
+                xmlw.writeCharacters("Different licenses and/or terms apply to individual files in the dataset. Access to some files in the dataset is restricted.");
+            } else {
+                xmlw.writeCharacters("Different licenses and/or terms apply to individual files in the dataset.");
+            }
+
+            xmlw.writeEndElement();
+        }
+    }
+
+    private static boolean areAllFilesUnderSameLincese(List<FileDTO> files) {
+        final String firstFileLicense = files.get(0).getLicenseName();
+        if(StringUtils.isNotEmpty(firstFileLicense)) {
```

Comment:
By looking at the entity you can see that license name cannot be empty.

Reply:
This is not from entity but from FileDTO, where it's being added with `.add("licenseName", Try.of(() -> fmd.getTermsOfUse().getLicense().getName()).getOrElse(StringUtils.EMPTY))` so it can be empty.

Reply:
Yeah but in the entity it says that it is not nullable (ofc you can set empty string), I would say that we should throw an exception "getOrElseThrow", since I don't think we allow the situation where license is empty. It should always have a name and it would be major error if it didn't.

Reply:
Not all files has license. Only those with `TermsOfUseType.LICENSE_BASED` has one. It's not an error if a file doesn't have a license.

Reply:
Ok then mistake is somewhere else since  ``` fmd.getTermsOfUse().getLicense().getName() ```
asumes all fmd has license, if it won't have one there will be null expection thrown. Since the getOrElse is only for the last chain method called.

Reply:
Oh yeah and there is no reason to use Try, it is only for exceptions, you should use optional or option if you wanna check if value is present.